ngraph_bridge/enable_variable_ops/ngraph_replace_op_utilities.cc

```diff
@@ -27,52 +27,47 @@ namespace tensorflow {
+                        const string replacement_node_name,
+                        const string replacement_node_type,
+                        const bool just_looking, const bool is_tf_just_looking,
+                        const bool outputs_ng_supported, const int graph_id,
+                        const bool is_backend_set) {
+  NGRAPH_VLOG(1) << "Start replacing " << node->type_string() << " "
                  << node->name();
 
-  DataType dtype;
-  TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "T", &dtype));
-  bool use_locking;
-  TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "use_locking", &use_locking));
-
-  NodeBuilder::NodeOut input_var;
-  NodeBuilder::NodeOut input_alpha;
-  NodeBuilder::NodeOut input_delta;
-
+  std::vector<NodeBuilder::NodeOut> op_inputs;
   std::vector<const Edge*> input_edges;
   TF_RETURN_IF_ERROR(node->input_edges(&input_edges));
 
-  NGRAPH_VLOG(1) << "No of input edges to ApplyGradientDescent "
-                 << input_edges.size();
+  NGRAPH_VLOG(1) << "No of input edges to Optimizer" << node->type_string()
+                 << " " << input_edges.size();
 
-  input_var =
-      NodeBuilder::NodeOut(input_edges[0]->src(), input_edges[0]->src_output());
-  input_alpha =
-      NodeBuilder::NodeOut(input_edges[1]->src(), input_edges[1]->src_output());
-  input_delta =
-      NodeBuilder::NodeOut(input_edges[2]->src(), input_edges[2]->src_output());
+  int num_inputs = node->num_inputs();
+  for (int i = 0; i < num_inputs; i++) {
+    op_inputs.push_back(NodeBuilder::NodeOut(input_edges[i]->src(),
+                                             input_edges[i]->src_output()));
+  }
 
-  TF_RETURN_IF_ERROR(NodeBuilder(replacement_node_name, replacement_node_type)
-                         .Attr("T", dtype)
-                         .Attr("use_locking", use_locking)
-                         .Attr("just_looking", just_looking)
-                         .Attr("is_tf_just_looking", is_tf_just_looking)
-                         .Attr("copy_to_tf", !outputs_ng_supported)
-                         .Attr("ngraph_graph_id", graph_id)
-                         .Input(input_var)
-                         .Input(input_alpha)
-                         .Input(input_delta)
-                         .Device(node->assigned_device_name())
-                         .Finalize(graph, &(*replacement)));
+  NodeBuilder nb = NodeBuilder(replacement_node_name, replacement_node_type)
+                       .Attr("just_looking", just_looking)
+                       .Attr("is_tf_just_looking", is_tf_just_looking)
+                       .Attr("copy_to_tf", !outputs_ng_supported)
+                       .Attr("ngraph_graph_id", graph_id)
+                       .Device(node->assigned_device_name());
+
+  for (auto it = node->attrs().begin(); it != node->attrs().end(); it++) {
+    nb.Attr(it->first, it->second);
+  }
 
+  for (auto const i : op_inputs) {
+    NGRAPH_VLOG(3) << "op_inputs is not empty";
```

Comment:
This prints the same thing in a loop. Maybe we could print `i` here, or move this outside the loop